util/argo/argo_test.go

```diff
@@ -161,8 +161,9 @@ func TestValidateChartWithoutRevision(t *testing.T) {
 		},
 	}, nil)
 	assert.NoError(t, err)
-	assert.ElementsMatch(t, conditions, []argoappv1.ApplicationCondition{{
-		Type: argoappv1.ApplicationConditionInvalidSpecError, Message: "spec.source.targetRevision is required if the manifest source is a helm chart"}})
+	assert.Equal(t, 1, len(conditions))
+	assert.Equal(t, argoappv1.ApplicationConditionInvalidSpecError, conditions[0].Type)
+	assert.Equal(t, "spec.source.targetRevision is required if the manifest source is a helm chart", conditions[0].Message)
```

Comment:
check time here?

Reply:
In current form I tried to steer away from setting the timestamp in any code that was going through app.Status.SetConditions and let that handle the logic of applying or updating the time. This function ended up no longer setting timestamp. Let me know what you think.